tools/python_api/src_py/prepared_statement.py

```diff
@@ -0,0 +1,68 @@
+class PreparedStatement:
+    """
+    A prepared statement is a parameterized query which can avoid planning the 
+    same query for repeated execution.
+
+    Methods
+    -------
+
+    is_success()
+        Check if the prepared statement is successfully prepared.
+
+    get_error_message()
+        Get the error message if the query is not prepared successfully.
+
+    is_read_only()
+        Check if the prepared statement is read-only.
+    """
+
+    def __init__(self, connection, query):
+        """
+        Parameters
+        ----------
+        connection : Connection
+            Connection to a database.
+        query : str
+            Query to prepare.
+        """
+
+        self._prepared_statement = connection._connection.prepare(query)
+        # Keep a reference to the connection to prevent make sure the connection
+        # is not garbage collected before the prepared statement.
+        self.connection = connection
```

Comment:
As we discussed, this may not be needed.

Reply:
I removed this and the tests still pass, so this is not needed